crates/jstzd/src/task/octez_client.rs

```diff
@@ -0,0 +1,152 @@
+use super::{endpoint::Endpoint, octez_node::OctezNode};
+use std::path::PathBuf;
+use tempfile::{tempdir, TempDir};
+
+pub struct OctezClientConfig {
+    binary_path: Option<PathBuf>,
+    // if None, use temp directory
+    base_dir: Option<PathBuf>,
+    endpoint: OctezNodeEndpoint,
+    disable_unsafe_disclaimer: bool,
+}
+
+impl OctezClientConfig {
+    pub fn new(endpoint: OctezNodeEndpoint) -> Self {
```

Comment:
Why must `new` take `endpoint`? If it's because of having to a default value for `endpoint`, the client binary by default sets the RPC endpoint to localhost:8732. We can probably follow this. What do you think?

Reply:
Do you mean asking users to call something like `OctezClientConfig::new(OctezNodeEndpoint::RpcEndpoint("localhost:8732"))` anyway?

Reply:
nono, i meant implemenitng the default for OctezNodeEndpoint that will return `localhost:8732`. the client don't need to specify anything

Reply:
Yes that sounds good